CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -105,10 +105,10 @@ public class AccountService : IAccountService
                         AccountId = user.Id
```

Comment:
Please move that newSecretary creation under the 'if' statement. And the same for other cases.